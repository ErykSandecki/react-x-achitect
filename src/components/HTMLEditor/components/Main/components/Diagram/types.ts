// types
import { BlockType, PortType, T2DCoordinates, TBlock, TConnector } from 'types';

export type TDraggableConnectorSource = {
  blockId: string;
  blockType: BlockType;
  portType: PortType;
};

export type TDraggableConnectorTarget = {
  blockId: string;
  blockType: BlockType;
  portType: PortType;
};

export type TDraggableConnector = {
  showIndicator: boolean;
  source?: TDraggableConnectorSource;
  sourceCoordinates: T2DCoordinates;
  target?: TDraggableConnectorTarget;
  targetCoordinates: T2DCoordinates;
};

export type TPossiblePort = T2DCoordinates & {
  blockId: string;
  blockType: BlockType;
  height: number;
  portType: PortType;
  width: number;
};

export type TShadowConnector = Pick<
  TConnector,
  'joinOperator' | 'relatedWithBlockId' | 'towardsTo'
> & {
  sourceBlock: Partial<TBlock> & { portType: PortType };
  targetBlock: Partial<TBlock> & { portType: PortType };
};
