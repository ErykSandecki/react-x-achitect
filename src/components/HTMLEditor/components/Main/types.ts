import { MutableRefObject } from 'react';

export type TMainRefs = {
  blocksPanelRef: MutableRefObject<HTMLDivElement>;
  diagramRef: MutableRefObject<HTMLDivElement>;
  panelPropertiesRef: MutableRefObject<HTMLDivElement>;
};
