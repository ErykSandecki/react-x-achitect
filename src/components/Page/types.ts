import { ReactNode } from 'react';

// types
import { PageType } from './enums';

export type TPageType = Record<PageType, string>;

export type TRestProps<T> = {
  children: ReactNode;
  pageType?: T;
  translationNameSpace?: string;
  withBackground?: boolean;
};
