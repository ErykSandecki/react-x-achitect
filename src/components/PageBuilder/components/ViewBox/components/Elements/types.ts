// types
import { MouseMode } from 'components/PageBuilder/enums';
import { TElement } from 'types';

export type TElementProps = {
  className: string;
  id: TElement['id'];
  mouseMode: MouseMode;
  parentId: TElement['parentId'];
  type: TElement['type'];
};
