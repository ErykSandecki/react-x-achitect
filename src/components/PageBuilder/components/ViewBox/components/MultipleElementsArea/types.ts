// types
import { TElement, TRectCoordinates } from 'types';

export type TElements = {
  elementsCordinates: Array<{
    coordinates: TRectCoordinates;
    id: TElement['id'];
  }>;
  outline: TRectCoordinates;
};
