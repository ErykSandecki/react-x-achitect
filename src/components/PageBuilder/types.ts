// types
import { TRectCoordinates } from 'types';

export type TRectArea = TRectCoordinates | null;
