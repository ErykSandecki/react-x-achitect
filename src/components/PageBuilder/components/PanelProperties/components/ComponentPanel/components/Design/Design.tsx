import { FC } from 'react';

// components
import ColumnHeader from './components/ColumnHeader/ColumnHeader';
import { UITools } from 'shared';

const Design: FC = () => {
  return (
    <>
      <UITools.Section>
        <ColumnHeader />
      </UITools.Section>
    </>
  );
};

export default Design;
