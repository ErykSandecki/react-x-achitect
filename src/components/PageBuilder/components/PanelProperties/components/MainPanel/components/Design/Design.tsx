import { FC } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useTranslation } from 'react-i18next';

// components
import { Icon, UITools } from 'shared';

// others
import { translationNameSpace } from './constants';

// store
import { changeBackground } from 'store/pageBuilder/actions';
import { pageBackgroundSelectorCreator } from 'store/pageBuilder/selectors';

export type TDesign = {};

const Design: FC<TDesign> = () => {
  const dispatch = useDispatch();
  const { alpha, format, value, visible } = useSelector(
    pageBackgroundSelectorCreator('-1'),
  );
  const { t } = useTranslation();

  return (
    <>
      <UITools.Section label={t(`${translationNameSpace}.section.1.label`)}>
        <UITools.SectionColumn
          buttonIcon={
            <UITools.ButtonIcon
              name={visible ? 'EyesOpened' : 'EyesClosed'}
              onClick={() =>
                dispatch(changeBackground({ visible: !visible }, '-1'))
              }
            />
          }
        >
          <UITools.ColorPicker
            alpha={alpha}
            color={value}
            format={format}
            onChangeAlpha={(alpha) =>
              dispatch(changeBackground({ alpha }, '-1'))
            }
            onChangeColor={(alpha, value) =>
              dispatch(changeBackground({ alpha, value }, '-1'))
            }
            onFormatChange={(format) =>
              dispatch(changeBackground({ format }, '-1'))
            }
            placement="leftBottom"
          />
        </UITools.SectionColumn>
      </UITools.Section>
    </>
  );
};

export default Design;
