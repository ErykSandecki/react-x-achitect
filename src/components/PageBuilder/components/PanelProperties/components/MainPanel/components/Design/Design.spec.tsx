import { fireEvent, render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';

// components
import Design from './Design';

// mocks
import { pageBuilderStateMock } from 'test/mocks/reducer/pageBuilderMock';

// others
import { REDUCER_KEY as PAGE_BUILDER } from 'store/pageBuilder/actionsType';

// store
import { configureStore } from 'store/store';

// types
import { E2EAttribute } from 'types';

// utils
import { getByE2EAttribute } from 'test';

const stateMock = {
  ...pageBuilderStateMock,
};

describe('Design snapshots', () => {
  it('should render Design', () => {
    // mock
    const store = configureStore(stateMock);

    // before
    const { asFragment } = render(
      <Provider store={store}>
        <Design />
      </Provider>,
    );

    // result
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('Design behaviors', () => {
  it('should change background visibility', () => {
    // mock
    const store = configureStore(stateMock);

    // before
    const { container } = render(
      <Provider store={store}>
        <Design />
      </Provider>,
    );

    // action
    fireEvent.click(
      getByE2EAttribute(container, E2EAttribute.icon, 'EyesOpened'),
    );

    // result
    expect(
      store.getState()[PAGE_BUILDER].pages['0'].elements.allData['-1']
        .background.visible,
    ).toBe(false);
  });

  it('should change alpha', async () => {
    // mock
    const store = configureStore(stateMock);

    // before
    const { container } = render(
      <Provider store={store}>
        <Design />
      </Provider>,
    );

    // find
    const input = getByE2EAttribute(
      container,
      E2EAttribute.textFieldInput,
      'alpha',
    );

    // action
    fireEvent.change(input, {
      target: { value: '50' },
    });
    fireEvent.blur(input);

    // result
    expect(
      store.getState()[PAGE_BUILDER].pages['0'].elements.allData['-1']
        .background.alpha,
    ).toBe('50');
  });

  it('should change color', async () => {
    // mock
    const store = configureStore(stateMock);

    // before
    const { container } = render(
      <Provider store={store}>
        <Design />
      </Provider>,
    );

    // find
    const input = getByE2EAttribute(
      container,
      E2EAttribute.textFieldInput,
      'color',
    );

    // action
    fireEvent.change(input, {
      target: { value: 'ffffff' },
    });
    fireEvent.blur(input);

    // result
    expect(
      store.getState()[PAGE_BUILDER].pages['0'].elements.allData['-1']
        .background.value,
    ).toBe('#ffffff');
  });

  it('should change format', async () => {
    // mock
    const store = configureStore(stateMock);

    // before
    const { container } = render(
      <Provider store={store}>
        <Design />
      </Provider>,
    );

    // action
    fireEvent.click(getByE2EAttribute(container, E2EAttribute.color));

    // find
    const dropdown = await screen.findByRole('combobox');

    // action
    await fireEvent.mouseDown(dropdown);

    // find
    const rgbOption = await screen.findByText('RGB');

    // action
    fireEvent.click(rgbOption);

    // result
    expect(
      store.getState()[PAGE_BUILDER].pages['0'].elements.allData['-1']
        .background.format,
    ).toBe('rgb');
  });
});
