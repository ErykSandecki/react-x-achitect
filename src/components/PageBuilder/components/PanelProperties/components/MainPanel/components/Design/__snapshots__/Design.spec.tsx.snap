// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Design snapshots should render Design 1`] = `
<DocumentFragment>
  <div
    class="css-q0izro Section Section--light"
    data-test-section=""
  >
    <small
      class="css-q0izro Section__label Section__label--light Typography Typography--light Typography--neutral1 Typography--neutral1--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
      data-test-text=""
      style="text-align: inherit;"
    >
      Page
    </small>
    <div
      class="css-q0izro SectionColumn SectionColumn--light"
      data-test-section=""
    >
      <div
        class="SectionColumn__content SectionColumn__content--light"
      >
        <div
          class="css-q0izro FieldGroup FieldGroup--light"
          data-test-field-group=""
        >
          <div
            class="css-q0izro ColorPicker__hex-input ColorPicker__hex-input--light TextField TextField--light"
            data-test-text-field="color"
          >
            <div
              aria-describedby="test-id"
            >
              <div
                class="css-q0izro Color Color--light"
                data-test-color=""
              >
                <div
                  class="Color__picker Color__picker--light"
                  style="background-color: rgb(241, 242, 246);"
                />
                <div
                  class="Color__picker-alpha Color__picker-alpha--light"
                  style="background-color: rgb(241, 242, 246);"
                />
                <div
                  class="Color__picker-texture Color__picker-texture--light"
                />
              </div>
            </div>
            <input
              class="TextField__input TextField__input--light"
              data-test-text-field-input="color"
              maxlength="6"
              value="f1f2f6"
            />
          </div>
          <div
            class="css-q0izro ColorPicker__alpha-input ColorPicker__alpha-input--light TextField TextField--light"
            data-test-text-field="alpha"
          >
            <input
              class="TextField__input TextField__input--light"
              data-test-text-field-input="alpha"
              max="100"
              maxlength="6"
              min="0"
              type="number"
              value="100"
            />
            <div
              class="ScrubbableInput ScrubbableInput--light"
              data-test-scrubbable-input=""
            >
              %
            </div>
          </div>
        </div>
      </div>
      <div
        class="SectionColumn__button SectionColumn__button--light"
      >
        <div
          class="css-q0izro ButtonIcon ButtonIcon--light"
          data-test-button-icon=""
        >
          <div
            class="css-q0izro Icon Icon--light Icon--neutral1 Icon--neutral1--light"
            data-test-icon="EyesOpened"
            height="14"
            name="EyesOpened"
            width="14"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
